src/components/MainSidebar/TimeDimensions.js

```diff
@@ -0,0 +1,81 @@
+import { DIMENSION_ID_PERIOD } from '@dhis2/analytics'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { useSelector } from 'react-redux'
+import {
+    TIME_DIMENSION_EVENT_DATE,
+    TIME_DIMENSION_ENROLLMENT_DATE,
+    TIME_DIMENSION_INCIDENT_DATE,
+    TIME_DIMENSION_SCHEDULED_DATE,
+    TIME_DIMENSION_LAST_UPDATED,
+    getTimeDimensions,
+    NAME_PARENT_PROPERTY_PROGRAM,
+} from '../../modules/timeDimensions.js'
+import { OUTPUT_TYPE_EVENT } from '../../modules/visualization.js'
+import { sGetMetadataById } from '../../reducers/metadata.js'
+import { sGetUiProgramId, sGetUiProgramStage } from '../../reducers/ui.js'
+import { DimensionsList } from './DimensionsList/index.js'
+import { PROGRAM_TYPE_WITH_REGISTRATION } from './ProgramDimensionsPanel/ProgramDimensionsPanel.js'
+
+const TimeDimensions = ({ selectedInputType }) => {
```

Comment:
Any special reason why `selectedInputType` needs to be passed instead of using a selector like the other ones below?

Reply:
Fixed